### DISCUSSION TITLE
add middleware and then request /docs occurs error

### DISCUSSION THREAD

**Discussion by LuckyCurve** [fastapi/fastapi#11866] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from datetime import time

import fastapi
from fastapi.requests import Request

app = fastapi.FastAPI()


@app.middleware("http")
async def add_process_time_header(request: Request, call_next):
    start_time = time.time()
    response = await call_next(request)
    process_time = time.time() - start_time
    response.headers["X-Process-Time"] = str(process_time)
    return response
```

### Description

run `fastapi run main.py` and open a broswer and call endpoints /docs will see Internal Server Error, how can i solve it?

### Operating System

Windows

### Operating System Details

windows version: 22H2
inner version:	19045.4651

### FastAPI Version

0.111.1

### Pydantic Version

2.8.2

### Python Version

Python 3.12.4

**Comment by YuriiMotov** [comment, marked as answer]
The problem is that you import `time` incorrectly.

Use `import time` instead of `from datetime import time`

**Reply by LuckyCurve** [reply]
thanks a lot, i made a mistake and i solve it with your answer
